test/command/5517.md

```diff
@@ -0,0 +1,28 @@
+Use epub output to trigger `downloadOrRead` in `Text.Pandoc.Class`
+in order to test `file:` URL-scheme handling.
+
+There are no relative `file:` URLs, so we cannot
+test with an actual file, since we don't know the
+current working directory. Instead, we redirect
+stderr to stdout and check for the correct warning.
+
+```
+% pandoc -M title=test -f native -t epub -o /dev/null 2>&1
+[Para [Image ("",[],[]) [] ("file:/testpath/foo.jpg","")]]
+^D
+[WARNING] Could not fetch resource 'file:/testpath/foo.jpg': PandocIOError "/testpath/foo.jpg" /testpath/foo.jpg: openBinaryFile: does not exist (No such file or directory)
+```
```

Comment:
I don't like the idea of triggering a read to an unknown location on the file system, even though a read should generally be harmless.

A cleaner way to test this feature (though more work to set up, since it would have to be done in Haskell) would be to use `runPure` and set up a fake file system.  (This is done using `modifyPureState` to update `stFiles` to a `FileTree` containing the files you want.  See also `insertInFileTree`.)

Reply:
Agreed that this would be a tad cleaner. Although it would open up the possibility of the fake fs behaving differently than the real one. Then again, we have Windows-specific preprocessor changes in the filepath handling anyway, which makes it difficult to test behaviour on different platforms...

I may have the time to look into this at some later point... but what do you think of reading from `/dev/null` for now? I updated the pull and it seems to work.

Reply:
OK, I guess that's good for now.